File: cs/libldt3.tests/libldt3/model/regel/F002Test.cs
using System;
using NUnit.Framework;

namespace libldt3
{
	namespace model
	{
		namespace regel
		{

			[TestFixture]
			public class F002Test
			{

				static readonly F002 f002 = new F002();

				[Test]
				public void testEmpty()
				{
					Assert.That(f002.IsValid(""), Is.False);
				}

				[Test]
				public void testNull()
				{
					Assert.That(f002.IsValid(null), Is.True);
				}

				[Test]
				public void testOnlyZeros()
				{
					Assert.That(f002.IsValid("00000000"), Is.False);
				}

				[Test]
				public void testFirstOfJanuaryEver()
				{
					Assert.That(f002.IsValid("00000101"), Is.True);
				}

				[Test]
				public void testLastOfDecemberEver()
				{
					Assert.That(f002.IsValid("99991231"), Is.True);
				}

				[Test]
				public void testDateOfTestCreation()
				{
					Assert.That(f002.IsValid("20161023"), Is.True);
				}

				[Test]
				public void testToLargeMonth()
				{
					Assert.That(f002.IsValid("20161323"), Is.False);
				}

				[Test]
				public void testMaxMonth()
				{
					Assert.That(f002.IsValid("20169923"), Is.False);
				}

				[Test]
				public void testToSmallMonth()
				{
					Assert.That(f002.IsValid("20160023"), Is.False);
				}

				[Test]
				public void testToLargeDay()
				{
					Assert.That(f002.IsValid("20161032"), Is.False);
				}

				[Test]
				public void testMaxDay()
				{
					Assert.That(f002.IsValid("20161099"), Is.False);
				}

				[Test]
				public void testToSmallDay()
				{
					Assert.That(f002.IsValid("20161000"), Is.False);
				}

				[Test]
				public void testToShort()
				{
					Assert.That(f002.IsValid("2016102"), Is.False);
				}

				[Test]
				public void testToLong()
				{
					Assert.That(f002.IsValid("201610234"), Is.False);
				}

				[Test]
				public void testAlphaOnly()
				{
					Assert.That(f002.IsValid("ABDCEFGH"), Is.False);
				}

				[Test]
				public void testAlphanumeric()
				{
					Assert.That(f002.IsValid("2O161O23"), Is.False);
				}

				[Test]
				public void testRandomString()
				{
					Assert.That(f002.IsValid(Guid.NewGuid().ToString()), Is.False);
				}
			}
		}
	}
}

File: java/src/test/java/libldt3/model/regel/test/F002Test.java
package libldt3.model.regel.test;

import libldt3.model.regel.format.F002;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.UUID;

public class F002Test {

	private static final F002 f002 = new F002();

	@Test
	public void testEmpty() {
		Assertions.assertFalse(f002.isValid(""));
	}

	@Test
	public void testNull() {
		Assertions.assertTrue(f002.isValid(null));
	}

	@Test
	public void testOnlyZeros() {
		Assertions.assertFalse(f002.isValid("00000000"));
	}

	@Test
	public void testFirstOfJanuaryEver() {
		Assertions.assertTrue(f002.isValid("00000101"));
	}

	@Test
	public void testLastOfDecemberEver() {
		Assertions.assertTrue(f002.isValid("99991231"));
	}

	@Test
	public void testDateOfTestCreation() {
		Assertions.assertTrue(f002.isValid("20161023"));
	}

	@Test
	public void testToLargeMonth() {
		Assertions.assertFalse(f002.isValid("20161323"));
	}

	@Test
	public void testMaxMonth() {
		Assertions.assertFalse(f002.isValid("20169923"));
	}

	@Test
	public void testToSmallMonth() {
		Assertions.assertFalse(f002.isValid("20160023"));
	}

	@Test
	public void testToLargeDay() {
		Assertions.assertFalse(f002.isValid("20161032"));
	}

	@Test
	public void testMaxDay() {
		Assertions.assertFalse(f002.isValid("20161099"));
	}

	@Test
	public void testToSmallDay() {
		Assertions.assertFalse(f002.isValid("20161000"));
	}

	@Test
	public void testToShort() {
		Assertions.assertFalse(f002.isValid("2016102"));
	}

	@Test
	public void testToLong() {
		Assertions.assertFalse(f002.isValid("201610234"));
	}

	@Test
	public void testAlphaOnly() {
		Assertions.assertFalse(f002.isValid("ABDCEFGH"));
	}

	@Test
	public void testAlphanumeric() {
		Assertions.assertFalse(f002.isValid("2O161O23"));
	}

	@Test
	public void testRandomString() {
		Assertions.assertFalse(f002.isValid(UUID.randomUUID().toString()));
	}

}
